02-express-tutorial/app.js

```diff
@@ -1,56 +1,27 @@
 const express = require("express");
-const { products } = require("./data");
+const cookieParser = require("cookie-parser");
 
-const app = express();
+const productRouter = require("./routes/product");
+const peopleRouter = require("./routes/people");
+const authenticationRouter = require("./routes/authentication");
 
-app.use(express.static("./public"));
+const app = express();
 
-app.get("/api/v1/test", (req, res) => {
-  res.json({ message: "It worked!" });
-});
+// Middleware to parse JSON
+app.use(express.json());
 
-app.get("/api/v1/products", (req, res) => {
-  res.json(products);
-});
+// Middleware to apply cookie-parser
+app.use(cookieParser());
 
-app.get("/api/v1/products/:productID", (req, res) => {
-  const prodID = parseInt(req.params.productID);
-  const result = products.find((product) => product.id === prodID);
-  result
-    ? res.json(result)
-    : res.status(404).json({ message: "That product was not found." });
-});
+// Middleware to server static files
+app.use(express.static("./public"));
 
-app.get("/api/v1/query", (req, res) => {
-  let { search, limit, price } = req.query;
-  let result = [...products];
-  // example: /api/v1/query?search=al
-  if (search) {
-    // example: /api/v1/query?search=/sec/
-    if (search.startsWith("/")) {
-      // the returned query string is in string format, it is converted to a regex object and trailing slashes are removed
-      search = new RegExp(search.slice(1, -1));
-      result = result.filter((product) => search.test(product.name));
-    } else {
-      result = result.filter((product) => product.name.startsWith(search));
-    }
-  }
-  // example: /api/v1/query?price=gt:10
-  if (price) {
-    price = price.split(":");
-    if (price[0] === "gt") {
-      result = result.filter((product) => product.price > price[1]);
-    } else if (price[0] === "lt") {
-      result = result.filter((product) => product.price < price[1]);
-    }
-  }
-  // example: /api/v1/query?limit=3
-  if (limit) {
-    result = result.slice(0, Number(limit));
-  }
-  res.json(result);
-});
+// Applying all my imported 'routers' to my main server with the help of express 'use' method
+app.use(peopleRouter);
```

Comment:
For these lines, you want to provide a route as the first parameter. i.e.e `app.use("/api/v1/people", peopleRouter);`